wgpu-hal/src/dx12/command.rs

```diff
@@ -1272,13 +1273,185 @@ impl crate::CommandEncoder for super::CommandEncoder {
+                                            aabb.buffer
+                                                .expect("needs buffer to build")
+                                                .resource
+                                                .GetGPUVirtualAddress()
+                                                + (aabb.offset as u64 * aabb.stride)
+                                        },
+                                        StrideInBytes: aabb.stride,
+                                    },
+                                },
+                            },
+                        })
+                    }
+                    (
+                        Direct3D12::D3D12_RAYTRACING_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL,
+                        Direct3D12::D3D12_ELEMENTS_LAYOUT_ARRAY,
+                        Direct3D12::D3D12_BUILD_RAYTRACING_ACCELERATION_STRUCTURE_INPUTS_0 {
+                            pGeometryDescs: geometry_desc.as_ptr(),
+                        },
+                        geometry_desc.len() as u32,
+                    )
+                }
+            };
+            let acceleration_structure_inputs =
+                Direct3D12::D3D12_BUILD_RAYTRACING_ACCELERATION_STRUCTURE_INPUTS {
+                    Type: ty,
+                    Flags: conv::map_acceleration_structure_build_flags(
+                        descriptor.flags,
+                        Some(descriptor.mode),
+                    ),
+                    NumDescs: num_desc,
+                    DescsLayout: layout,
+                    Anonymous: inputs0,
+                };
+            let desc = Direct3D12::D3D12_BUILD_RAYTRACING_ACCELERATION_STRUCTURE_DESC {
+                DestAccelerationStructureData: unsafe {
+                    descriptor
+                        .destination_acceleration_structure
+                        .resource
+                        .GetGPUVirtualAddress()
+                },
+                Inputs: acceleration_structure_inputs,
+                SourceAccelerationStructureData: descriptor
+                    .source_acceleration_structure
+                    .as_ref()
+                    .map_or(0, |source| unsafe {
+                        source.resource.GetGPUVirtualAddress()
+                    }),
+                ScratchAccelerationStructureData: unsafe {
+                    descriptor.scratch_buffer.resource.GetGPUVirtualAddress()
+                        + descriptor.scratch_buffer_offset
+                },
+            };
+            unsafe { list.BuildRaytracingAccelerationStructure(&desc, None) };
+        }
     }
 
     unsafe fn place_acceleration_structure_barrier(
         &mut self,
         _barriers: crate::AccelerationStructureBarrier,
     ) {
-        todo!()
+        // TODO: This is not very optimal, we should be using [enhanced barriers](https://microsoft.github.io/DirectX-Specs/d3d/D3D12EnhancedBarriers.html) if possible
```

Comment:
Could you file a separate issue for using enhanced barriers. Unfortunately we can't unconditionally support it.

Reply:
opened #6829